server/src/test/java/keywhiz/service/resources/automation/v2/SecretResourceTest.java

```diff
@@ -275,6 +277,87 @@
     assertThat(s3).doesNotContain("secret17");
   }
 
+  @Test public void secretVersionListing_notFound() throws Exception {
+    Request put = clientRequest("/automation/v2/secrets/non-existent/versions/0-0").build();
+    Response httpResponse = mutualSslClient.newCall(put).execute();
+    assertThat(httpResponse.code()).isEqualTo(404);
+  }
+
+  @Test public void secretVersionListing_success() throws Exception {
+    int totalVersions = 6;
+    int sleepInterval = 1000; // Delay so secrets have different creation timestamps
```

Comment:
Normally in this situation I'd use a fake clock, but that appears to not exist in this world.

It looks like in general we don't have great things to test for timestamps correctly (and adding that is out of the scope of this PR). I think I would just manually munge the timestamps in SQL if possible to get around this for now, and open an issue to get better test coverage of timestamps for the future.